### PR TITLE
bump hugo-bin version v0.123.0

### PR DESCRIPTION
**Pull request by StefonSimmons**
### What does this PR do? What is the motivation?

bumps hugo-bin

NOTE: PRs to remove `kind` from external sources (these are non-blocking for this PR):
- https://github.com/DataDog/apps/pull/169
- https://github.com/DataDog/datadog-serverless-functions/pull/814
- https://github.com/DataDog/datadog-agent/pull/27205 

### Merge instructions
- [ ] Please merge after webops reviews

### Additional notes
Full site QA needed due to Hugo version bump: https://docs-staging.datadoghq.com/stefon.simmons/bump-hugo-bin-version/

- images should be loading
- resource links should work, ex https://docs-staging.datadoghq.com/stefon.simmons/bump-hugo-bin-version/containers/amazon_ecs/?tab=awscli#create-and-manage-the-task-definition-file (check the json file link)
- spot check JS functionality (tabs, left nav, search, etc)
- spot check various pages to make sure there are no issues.

We will re-enable the preview cache flag after QA is completed.

### CODE REVIEW

**Review by domalessi — APPROVED** [review]

**Review by davidejones — CHANGES_REQUESTED** [review]
I've noticed a few things:

- The [404 page](https://docs-staging.datadoghq.com/stefon.simmons/bump-hugo-bin-version/404.html) has no left nav 
- I've also noticed a number of urls now have the anchor in the href or query params in it, this doesn't appear to be causing any issues but thought i should call it out
Screenshot 2024-07-10 at 10 02 18
- Some images aren't working e.g see these pages [here](https://docs-staging.datadoghq.com/stefon.simmons/bump-hugo-bin-version/service_management/events/) and [here](https://docs-staging.datadoghq.com/stefon.simmons/bump-hugo-bin-version/service_management/case_management/)

**Comment by bgdeutsch** [comment]
> I've noticed a few things:
> 
> * An update with master locally meant i also needed to update en/agent/guide to remove some more kind entries
> * The [404 page](https://docs-staging.datadoghq.com/stefon.simmons/bump-hugo-bin-version/404.html) has no left nav
> * Some images aren't working e.g see these pages [here](https://docs-staging.datadoghq.com/stefon.simmons/bump-hugo-bin-version/service_management/events/) and [here](https://docs-staging.datadoghq.com/stefon.simmons/bump-hugo-bin-version/service_management/case_management/)
> * I've also noticed a number of urls now have the anchor in the href or query params in it, this doesn't appear to be causing any issues but thought i should call it out

thanks @davidejones!
- there may still be some commits containing `kind` from translations or other changes.   that param is mostly gone from source files in transifex, and any working branches in docs containing `kind` will just have to be updated after this PR goes out.
- 404 page should be showing the left nav now
- images should be working again
- the URLs containing anchor or query params doesnt seem to have any effect as you mentioned.

**Review by davidejones — APPROVED** [review]

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 - config:
-    cache_enabled: true
+    cache_enabled: false
```
Comment (hestonhoffman) [inline local/bin/py/build/configurations/pull_config_preview.yaml:3]: Need to switch this back?

**Review by hestonhoffman — APPROVED** [review]
Poked around in preview - LGTM, but I think the cache value needs to be switched back?

**Review by domalessi — APPROVED** [review]

**Review by hestonhoffman — COMMENTED** [review]
local build is also failing for me on:
Error: error building site: process: readAndProcessContent: "/Users/heston.hoffman/repos/documentation/content/en/code_analysis/static_analysis_rules/csharp-best-practices/avoid-call-gc-suppress-finalize.md:1:1": unknown kind "documentation" in front matter
Weird that it's not happening in CI. I did a clean-all before building.
Does the logic that removes the kind parameter need to be in the pull-and-push-folder action as well?

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**